jedi/api/__init__.py

```diff
@@ -108,7 +110,11 @@ def __init__(self, source=None, line=None, column=None, path=None,
         self._user_context = UserContext(self.source, self._pos)
         self._parser = UserContextParser(self._grammar, self.source, path,
                                          self._pos, self._user_context)
-        self._evaluator = Evaluator(self._grammar)
+        if sys_path is None:
+            venv = os.getenv('VIRTUAL_ENV')
+            if venv:
+                sys_path = list(get_venv_path(venv)) + sys.path
```

Comment:
> Can you quote yourself from earlier how this PR improves virtualenvs?

- fetching `sys_path` from a live interpreter in the specified virtualenv, if possible
- if online fetching fails, offline detector now additionally follows `.pth` links (cannot seem to find the `addsitedir` invocation in the old code)
- `egg-link` files are consulted in all new path directories, not just the `site-packages` one